src/workerd/api/BUILD.bazel

```diff
@@ -38,6 +44,8 @@ api_header_exclusions = [
     "html-rewriter.h",
     "deferred-proxy.h",
     "data-url.h",
+    "encoding.h",
+    "util.h",
     "modules.h",
```

Comment:
If a source file is being excluded somewhere, the corresponding header should also be excluded from the target, otherwise it's easy to get linker errors due to missing symbols (e.g. when depending on io but not on util-iocontext). I think the goal of splitting util.c++ to move it out of the io target is worth pursing, but either util-iocontext.c++ must be a part of the io target; or the one function it has should be moved into io-context.c++/.h itself (second option is cleaner IMHO).